test/nn/models/test_g_retriever.py

```diff
@@ -0,0 +1,41 @@
+import torch
+
+from torch_geometric.data import Data
+from torch_geometric.nn.models import GRetriever
+from torch_geometric.testing import onlyFullTest, withPackage
+
+
+@onlyFullTest
+@withPackage('transformers', 'sentencepiece', 'accelerate')
+def test_g_retriever() -> None:
+    batch = Data()
+    batch.question = ["Is PyG the best open-source GNN library?"]
+    batch.label = ["yes!"]
+    batch.num_nodes = 10
+    batch.n_id = torch.arange(10).to(torch.int64)
+    batch.num_edges = 20
+    batch.x = torch.randn((10, 1024))
+    batch.edge_attr = torch.randn((20, 1024))
+    # Model expects batches sampled from Dataloader
+    # hardcoding values for single item batch
+    batch.batch = torch.zeros(batch.num_nodes).to(torch.int64)
+    batch.ptr = torch.tensor([0, batch.num_nodes]).to(torch.int64)
+    # simple cycle graph
+    batch.edge_index = torch.tensor(
+        [list(range(10)), list(range(1, 10)) + [0]])
+
+    model = GRetriever(
+        llm_to_use="TinyLlama/TinyLlama-1.1B-Chat-v0.1",
+        num_llm_params=1,  # 1 Billion
+        mlp_out_dim=2048,
+    )
```

Comment:
What would be nice is to mock the HF code that `GRetriever` is calling so that we don't have to download anything for testing.

Reply:
the problem is that huggingface code under the hood changes often, it will be hard to mantain a mirror of that implementation just for testing purposes. in general thats why we chose to use huggingface as the LLM instead of implementing our own, open source LLM engineering progresses far quicker than we can keep up with for GNN+LLM, which is a smaller community. On top of this, the tiny LLM used in testing is not likely to be the same LLM used in production settings where this code will be ultimately used. the testing for huggingface this way still tests the underlying transformers backbone they use.